requests/utils.py

```diff
@@ -453,13 +452,27 @@ def get_encoding_from_headers(headers):
     :param headers: dictionary to extract encoding from.
     :rtype: str
     """
+    def parse_header(content_type):
+        #Inner function to parse header
+        #append delimiter on end to ensure atleast two elements when split by ';'
+        content_type_and_params_delimiter = ';'
+        content_type += content_type_and_params_delimiter
+
+        tokens = content_type.split(content_type_and_params_delimiter)
+        content_type_index = 0
+        params_index = 1
+
+        content_type = tokens[content_type_index]
+        params = tokens[params_index]
+        params_dict = dict(param.split('=') for param in params.split())
```

Comment:
I'm not convinced this is going to be correct. I think having an `=` inside the parameter value is acceptable if it's quoted. We should probably then use `param.split('=', 1)` to ensure we only split on the first found =.

Reply:
Ill check it out. Thank you